src/twisted/test/test_tcp.py

```diff
@@ -118,9 +119,9 @@ class MyProtocolFactoryMixin:
 
     protocolFactory = AccumulatingProtocol
 
-    protocolConnectionMade = None
-    protocolConnectionLost = None
-    protocol: Optional[Callable[[], Protocol]] = None
+    protocolConnectionMade: Optional[defer.Deferred[AccumulatingProtocol]] = None
+    protocolConnectionLost: Optional[defer.Deferred[AccumulatingProtocol]] = None
+    createdProtocol: Optional[AccumulatingProtocol] = None
```

Comment:
Can you please also  update the docstring at line 104

Maybe this can be called something light this ... or `latestProtocol` or `recentProtocol`

```suggestion
    lastProtocol: Optional[AccumulatingProtocol] = None
```

Reply:
Fixed now.